beep/structure/cli.py

```diff
@@ -0,0 +1,271 @@
+import json
+
+from docopt import docopt
+from monty.serialization import loadfn, dumpfn
+
+from beep import logger, __version__
+from beep.conversion_schemas import (
+    FastCharge_CONFIG,
+    xTesladiag_CONFIG,
+    ARBIN_CONFIG,
+    MACCOR_CONFIG,
+    INDIGO_CONFIG,
+    BIOLOGIC_CONFIG,
+    NEWARE_CONFIG
+)
+from beep.structure.arbin import ArbinDatapath
+from beep.structure.maccor import MaccorDatapath
+from beep.structure.neware import NewareDatapath
+from beep.structure.indigo import IndigoDatapath
+from beep.structure.biologic import BiologicDatapath
+from beep.structure.base import BEEPDatapath
+from beep.utils import WorkflowOutputs
+from beep.collate import add_suffix_to_filename
+
+
+SERVICE_CONFIG = {"service": "DataStructurer"}
+
+
+def process_file_list_from_json(file_list_json, processed_dir="data-share/structure/"):
+    """Function to take a json filename corresponding to a data structure
+    with a 'file_list' and a 'validity' attribute, process each file
+    with a corresponding True validity, dump the processed file into
+    a predetermined directory, and return a jsonable dict of processed
+    cycler run file locations
+
+    Args:
+        file_list_json (str): json string or json filename corresponding
+            to a dictionary with a file_list and validity attribute,
+            if this string ends with ".json", a json file is assumed
+            and loaded, otherwise interpreted as a json string.
+        processed_dir (str): location for processed cycler run output
+            files to be placed.
+
+    Returns:
+        (str): json string of processed files (with key "processed_file_list").
+            Note that this list contains None values for every file that
+            had a corresponding False in the validity list.
+
+    """
+    # Get file list and validity from json, if ends with .json,
+    # assume it's a file, if not assume it's a json string
+    if file_list_json.endswith(".json"):
+        file_list_data = loadfn(file_list_json)
+    else:
+        file_list_data = json.loads(file_list_json)
+
+    # Setup workflow
+    outputs = WorkflowOutputs()
+
+    # Prepend optional root to output directory
+    processed_dir = os.path.join(
+        os.environ.get("BEEP_PROCESSING_DIR", "/"), processed_dir
+    )
+
+    if not os.path.exists(processed_dir):
+        os.makedirs(processed_dir)
+
+    file_list = file_list_data["file_list"]
+    validities = file_list_data["validity"]
+    run_ids = file_list_data["run_list"]
+    processed_file_list = []
+    processed_run_list = []
+    processed_result_list = []
+    processed_message_list = []
+    invalid_file_list = []
+    for filename, validity, run_id in zip(file_list, validities, run_ids):
+        logger.info("run_id=%s structuring=%s", str(run_id), filename, extra=SERVICE_CONFIG)
+        if validity == "valid":
+            # Process datapath and dump to file
+
+            dp = auto_load(filename)
+            dp.structure()
```

Comment:
@ardunn Should this be `autostructure`?

Reply:
Yes. I'll push a commit to fix that now.